src/features/handlers/json-rpc/JsonRpc.tsx

```diff
@@ -1,8 +1,72 @@
-import { Box, Heading } from '@chakra-ui/react';
+import {
+  Box,
+  Button,
+  Container,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+} from '@chakra-ui/react';
 import { FunctionComponent } from 'react';
+import { Controller, useForm } from 'react-hook-form';
 
-export const JsonRpc: FunctionComponent = () => (
-  <Box>
-    <Heading>JSON-RPC</Heading>
-  </Box>
-);
+import { Editor } from '../../../components';
+import { SAMPLE_JSON_RPC_REQUEST } from './schema';
+
+type JsonRpcFormData = {
+  origin: string;
+  request: string;
+};
+
+export const JsonRpc: FunctionComponent = () => {
+  const {
+    handleSubmit,
+    register,
+    control,
+    formState: { errors },
+  } = useForm<JsonRpcFormData>({
+    defaultValues: {
+      origin: '',
+      request: SAMPLE_JSON_RPC_REQUEST,
+    },
+  });
+
+  const onSubmit = (data: JsonRpcFormData) => {
+    console.log(data);
+  };
+
+  return (
+    <Box flex="1">
+      <Container size="fullWidth">
+        {/* eslint-disable-next-line @typescript-eslint/no-misused-promises */}
```

Comment:
?

Reply:
`handleSubmit` returns a `Promise`, but it's not handled by `onSubmit`. I'm not sure how to do it otherwise, this is how it's done in the docs.